tools/git.py

```diff
@@ -31,27 +33,44 @@ def __init__(
         :param commit: Git commit hash
         :param checkout_folder_name: Name of the output path repository folder
         :param clone_submodules: Flag to clone submodules if set to True
+        :param files_to_checkout: List of files to be checked out sparsely. If not set, the entire repo is checked out (default behaviour).
         """
         self.url = url
         self.branch = branch
         self.commit = commit
         self.clone_submodules = clone_submodules
+        self.files_to_checkout = files_to_checkout
 
         self.out_repository = self.output_path(checkout_folder_name, True)
 
     def tasks(self):
         yield Task("run", mini_task=True)
 
     def run(self):
-        args = ["git", "clone", self.url]
+        if self.files_to_checkout is not None:
+            args = ["git", "clone", "--no-checkout", self.url]
+        else:
+            args = ["git", "clone", self.url]
         if self.branch is not None:
             args.extend(["-b", self.branch])
         repository_dir = self.out_repository.get_path()
         args += [repository_dir]
         logging.info("running command: %s" % " ".join(args))
         sp.run(args, check=True)
 
-        if self.commit is not None:
+        if self.files_to_checkout is not None:
+            args = ["git", "checkout"]
+            commit = self.commit if self.commit is not None else ""
```

Comment:
In case that `self.commit` is not provided, I have tried the following:
```
$ git clone --no-checkout https://github.com/rwth-i6/i6_core.git
$ cd i6_core
$ git checkout -- .gitignore
```
The result is:
```
error: pathspec '.gitignore' did not match any file(s) known to git.
```
From experimenting, the correct checkout seems to be done by adding any branch name right after checkout, but it must be a valid one, like:
```
git checkout main -- .gitignore
```
Which properly tracks the `.gitignore` file. Therefore, the straightforward solution would be the following:
```suggestion
            commit = self.commit if self.commit is not None else "HEAD"
```

An alternative would be to clone the entire repository normally by not using `--no-checkout` above in order to have the whole repository available, and the condition to be checked in this conditional branch would be:
```
if self.files_to_checkout is not None and self.commit is not None:
```

Reply:
Requesting changes only because of this.

Reply:
Opos, I didn't test that condition - good that you did. I'll go for the first option as the second one would still check out the whole repo - something that should be avoided if files_to_checkout is set.